src/cloud_provider/helm.rs

```diff
@@ -591,47 +587,34 @@ impl HelmChart for PrometheusOperatorConfigChart {
+
         match chart_info.action {
             HelmAction::Deploy => {
-                if let Err(e) = helm_destroy_chart_if_breaking_changes_version_detected(
-                    kubernetes_config,
-                    &environment_variables,
-                    chart_info,
-                ) {
+                if let Err(e) = helm.uninstall_chart_if_breaking_version(chart_info, &vec![]) {
                     warn!(
                         "error while trying to destroy chart if breaking change is detected: {}",
-                        e.message()
+                        e.to_string()
                     );
                 }
 
-                helm_exec_upgrade_with_chart_info(kubernetes_config, &environment_variables, chart_info)?
+                helm.upgrade(&chart_info, &vec![]).map_err(to_command_error)?;
             }
             HelmAction::Destroy => {
                 let chart_info = self.get_chart_info();
-                match is_chart_deployed(
-                    kubernetes_config,
-                    environment_variables.clone(),
-                    Some(chart_info.get_namespace_string().as_str()),
-                    chart_info.name.clone(),
-                ) {
-                    Ok(deployed) => {
-                        if deployed {
-                            let prometheus_crds = [
-                                "prometheuses.monitoring.coreos.com",
-                                "prometheusrules.monitoring.coreos.com",
-                                "servicemonitors.monitoring.coreos.com",
-                                "podmonitors.monitoring.coreos.com",
-                                "alertmanagers.monitoring.coreos.com",
-                                "thanosrulers.monitoring.coreos.com",
-                            ];
-                            helm_exec_uninstall_with_chart_info(kubernetes_config, &environment_variables, chart_info)?;
-                            for crd in &prometheus_crds {
-                                kubectl_exec_delete_crd(kubernetes_config, crd, environment_variables.clone())?;
-                            }
-                        }
-                    }
-                    Err(e) => return Err(e),
-                };
+                helm.uninstall(&chart_info, &vec![]).map_err(to_command_error)?;
+
+                let prometheus_crds = [
+                    "prometheuses.monitoring.coreos.com",
+                    "prometheusrules.monitoring.coreos.com",
+                    "servicemonitors.monitoring.coreos.com",
+                    "podmonitors.monitoring.coreos.com",
+                    "alertmanagers.monitoring.coreos.com",
+                    "thanosrulers.monitoring.coreos.com",
+                ];
+                for crd in &prometheus_crds {
+                    let _ = kubectl_exec_delete_crd(kubernetes_config, crd, environment_variables.clone());
```

Comment:
don't you manage errors?

Reply:
We invoke it always now, so it returns an error if the crd is not present